File: LeetCode/1.Two_Sum/1.Two_Sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var dic :[Int:Int] = [:]
        for (i, num) in nums.enumerated(){
            if let idx = dic[target-num]{
                return [idx, i]
            }
            dic[num] = i
        }
        return []
    }
}

// test
let s = Solution()
let res = s.twoSum([2,7,11,15], 9)
print("res:\(res)")

File: LeetCode/1.Two_Sum/1.Two_Sum.go
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, num := range(nums) {
		index, ok := m[num]
		if ok {
			return []int{index, i}
		} else {
			m[target-num] = i
		}
	}
	return []int{0,0}
}